### DISCUSSION TITLE
Enabling pre-requisites in course consumption

### DISCUSSION THREAD

**Discussion by Krishnaj20** [Sunbird-Lern/Community#69] [Enhancement]
@rhwarrier 
Context: 
Currently, learners have the flexibility of consuming the course at their own pace and in the sequence they deem appropriate. We've seen an increasing market demand, to allow for multiple modes of course consumption. Although there are multiple modes, in this thread we are calling out the priority ones below:

a) allowing batch creator to be able to define pre-requisites for each course module. In terms of pre-requisites, the higher priority ask is to have: 
- users' completion of a module as a pre-requisite for unlocking the next module or 
- date when the module is to be made available for learners. 
There are other asks around score, time spent as pre-requisites but completion & date (called out above) are of higher priority.

Let us know if this item is already in your roadmap. Since this is a most asked feature we expect to see high usage. 

Please see 'https://github.com/orgs/Sunbird-ed/discussions/262' for more details.

@parthlawate

**Comment by rhwarrier** [comment]
@parthlawate  @mathewjpallan let's use this thread to discuss and close on design so that this can be addressed as part of the next release. 
@reshmi-nair please block some time to discuss this. Thanks

**Reply by parthlawate** [reply]
Sure. If you look at the original thread here https://github.com/orgs/Sunbird-ed/discussions/262 by @snehal0904 , she has proposed a design based on storage against a course. 

We are now proposing that we store this against a batch of a course to make it flexible i think ? I am good with that approach.

**Reply by reshmi-nair** [reply]
@rhwarrier @parthlawate I have blocked time on Mar 6th.
